### ISSUE TITLE
OpenJDK Windows build failed, No rule to make target jpp.jar

### ISSUE THREAD

**Issue by pshipton** [eclipse/openj9#11410]
https://ci.eclipse.org/openj9/job/Build_JDKnext_x86-64_windows_OpenJDK/357
```
3:05:37  Configuration summary:
13:05:37  * Name:           windows-x86_64-server-release
13:05:37  * Debug level:    release
13:05:37  * HS debug level: product
13:05:37  * JVM variants:   server
13:05:37  * JVM features:   server: 'cds compiler1 compiler2 epsilongc g1gc graal jfr jni-check jvmci jvmti management nmt parallelgc serialgc services shenandoahgc vm-structs zgc' 
13:05:37  * OpenJDK target: OS: windows, CPU architecture: x86, address length: 64
13:05:37  * Version string: 16-internal+0-adhoc.jenkins.buildjdknextx86-64windowsopenjdk (16-internal)
13:05:37  
13:05:37  Tools summary:
13:05:37  * Environment:    cygwin version 2.10.0(0.325/5/3), 2018-02-02 15:16; windows version 6.3.9600; prefix "/cygdrive"; root "C:\cygwin64"
13:05:37  * Boot JDK:       openjdk version "15" 2020-09-15 OpenJDK Runtime Environment AdoptOpenJDK (build 15+36) Eclipse OpenJ9 VM AdoptOpenJDK (build openj9-0.22.0, JRE 15 Windows Server 2012 R2 amd64-64-Bit Compressed References 20200922_45 (JIT enabled, AOT enabled) OpenJ9 - 1830b1927 OMR - 73d5e7623 JCL - 7e7613c015 based on jdk-15+36) (at /cygdrive/f/users/jenkins/bootjdks/jdk15)
13:05:37  * Toolchain:      microsoft (Microsoft Visual Studio 2019)
13:05:37  * C Compiler:     Version 19.26.28806 (at /cygdrive/c/progra~2/mib055~1/2019/commun~1/vc/tools/msvc/1426~1.288/bin/hostx86/x64/cl.exe)
13:05:37  * C++ Compiler:   Version 19.26.28806 (at /cygdrive/c/progra~2/mib055~1/2019/commun~1/vc/tools/msvc/1426~1.288/bin/hostx86/x64/cl.exe)
13:05:37  
13:05:37  Build performance summary:
13:05:37  * Cores to use:   7
13:05:37  * Memory limit:   8187 MB
13:05:37  
13:05:38  + make all
13:05:44  Building OpenJ9 Java Preprocessor
13:05:44  make[3]: *** No rule to make target 'f:\users\jenkins\workspace\build_jdknext_x86-64_windows_openjdk\build\windows-x86_64-server-release\support\j9tools\jpp.jar'.  Stop.
13:05:44  make[2]: *** [/cygdrive/f/users/jenkins/workspace/build_jdknext_x86-64_windows_openjdk/closed/GensrcJ9JCL.gmk:47: /cygdrive/f/users/jenkins/workspace/build_jdknext_x86-64_windows_openjdk/build/windows-x86_64-server-release/make-support/j9jcl.done] Error 2
13:05:44  Building OpenJ9 Java Preprocessor
13:05:45  make[3]: *** No rule to make target 'f:\users\jenkins\workspace\build_jdknext_x86-64_windows_openjdk\build\windows-x86_64-server-release\support\j9tools\jpp.jar'.  Stop.
13:05:45  make[2]: *** [/cygdrive/f/users/jenkins/workspace/build_jdknext_x86-64_windows_openjdk/closed/GensrcJ9JCL.gmk:47: /cygdrive/f/users/jenkins/workspace/build_jdknext_x86-64_windows_openjdk/build/windows-x86_64-server-release/make-support/j9jcl.done] Error 2
13:05:45  Building OpenJ9 Java Preprocessor
13:05:45  make[3]: *** No rule to make target 'f:\users\jenkins\workspace\build_jdknext_x86-64_windows_openjdk\build\windows-x86_64-server-release\support\j9tools\jpp.jar'.  Stop.
13:05:45  make[2]: *** [/cygdrive/f/users/jenkins/workspace/build_jdknext_x86-64_windows_openjdk/closed/GensrcJ9JCL.gmk:47: /cygdrive/f/users/jenkins/workspace/build_jdknext_x86-64_windows_openjdk/build/windows-x86_64-server-release/make-support/j9jcl.done] Error 2
13:05:45  make[1]: *** [/cygdrive/f/users/jenkins/workspace/build_jdknext_x86-64_windows_openjdk/closed/custom/common/Modules.gmk:52: generate-j9jcl-sources] Error 2
13:05:45  make: *** No rule to make target 'all'.  Stop.
```

**Comment by pshipton** [comment]
@tajila fyi this problem still blocking jdk16 acceptance.

I'm restarting the build just in case this problem is intermittent.

**Comment by pshipton** [comment]
@JasonFengJ9 fyi, maybe another place that needs fixing.

**Comment by pshipton** [comment]
The problem isn't intermittent.

**Comment by JasonFengJ9** [comment]
Will take a look.

**Comment by JasonFengJ9** [comment]
This was caused by `FixPath` change introduced via https://github.com/ibmruntimes/openj9-openjdk-jdk/commit/9bac781cde77b39e4d380d6d3ed71d8ef052805f
Before this change, `$(call FixPath,$(JPP_JAR))` is `C:/openj9-builds/jdk15-1208/openj9-openjdk-jdk15/build/windows-x86_64-server-release/support/j9tools/jpp.jar`.
With this change, `$(call FixPath,$(JPP_JAR))` is `c:\openj9-builds\jdk16-1208-staging\openj9-openjdk-jdk\build\windows-x86_64-server-release\support\j9tools\jpp.jar`
Note that the forward slash is replaced with backslash which caused error in this issue.
Adding quotation around the path doesn't help.

**Comment by JasonFengJ9** [comment]
A quick update, after updated a few directory separators, the jar files are built ok, now hit an OMR configuration error:
```
12:49:07  sh configure --disable-auto-build-flag 'OMRGLUE=../gc_glue_java' 'SPEC=win_x86-64_cmprssptrs'  --enable-OMR_OPT_CUDA --enable-OMRPORT_OMRSIG_SUPPORT --enable-OMR_GC --enable-OMR_PORT --enable-OMR_THREAD --enable-OMR_OMRSIG --enable-tracegen --enable-OMR_GC_DYNAMIC_CLASS_UNLOADING --enable-OMR_GC_MODRON_COMPACTION --enable-OMR_GC_MODRON_CONCURRENT_MARK --enable-OMR_GC_MODRON_SCAVENGER --enable-OMR_GC_CONCURRENT_SWEEP --enable-OMR_GC_SEGREGATED_HEAP --enable-OMR_GC_LEAF_BITS --enable-OMR_GC_REALTIME --enable-OMR_GC_VLHGC --enable-OMR_PORT_ASYNC_HANDLER --enable-OMR_THR_CUSTOM_SPIN_OPTIONS --enable-OMR_NOTIFY_POLICY_CONTROL --enable-debug --enable-DDR 'lib_output_dir=$(top_srcdir)/../lib' 'exe_output_dir=$(top_srcdir)/..' 'GLOBAL_INCLUDES=$(top_srcdir)/../include' --enable-debug --enable-OMR_THR_THREE_TIER_LOCKING --enable-OMR_THR_SPIN_WAKE_CONTROL --enable-OMRTHREAD_LIB_WIN32 --enable-OMR_ARCH_X86 --enable-OMR_ENV_DATA64 --enable-OMR_ENV_LITTLE_ENDIAN --enable-OMR_GC_TLH_PREFETCH_FTA --enable-OMR_GC_CONCURRENT_SCAVENGER --enable-OMR_PORT_ALLOCATE_TOP_DOWN --enable-OMR_PORT_CAN_RESERVE_SPECIFIC_ADDRESS OMR_GC_POINTER_MODE=compressed libprefix= exeext=.exe solibext=.dll arlibext=.lib objext=.obj 'AS=ml64' 'CC=cl' 'CXX=cl' 'CCLINK=link' 'CXXLINK=link' 'AR=lib' 'OMR_HOST_OS=win' 'OMR_HOST_ARCH=x86' 'OMR_TARGET_DATASIZE=64' 'OMR_TOOLCHAIN=msvc' 'OMR_COMPANY_NAME=International Business Machines Corporation' 'OMR_COMPANY_COPYRIGHT=Copyright (c) 1991, 2020 IBM Corp. and others' 'OMR_PRODUCT_NAME=IBM SDK, Java(tm) 2 Technology Edition' 'OMR_PRODUCT_DESCRIPTION=J9 Virtual Machine Runtime' 'OMR_PRODUCT_VERSION=16.0.0.0' 
12:49:50  Compiling 11 properties into resource bundles for jdk.jartool
12:49:56  Compiling 11 properties into resource bundles for jdk.management.agent
12:49:58  Compiling 71 files for COMPILE_CREATE_SYMBOLS
12:50:07  Compiling 4 properties into resource bundles for jdk.jlink
12:50:08  Compiling 3 properties into resource bundles for jdk.jlink
12:50:12  Compiling 3 properties into resource bundles for jdk.jdi
12:50:13  Compiling 1 properties into resource bundles for jdk.jlink
12:50:31  Creating javadoc element list
12:50:32  Compiling 224 properties into resource bundles for jdk.localedata
12:50:32  Compiling 71 files for COMPILE_CREATE_SYMBOLS
12:50:32  Compiling 11 properties into resource bundles for java.base
12:50:34  Compiling 6 properties into resource bundles for java.base
12:51:24  error: module not found: jdk.hotspot.agent
12:51:38  checking build system type... Creating ct.sym classes
12:51:42  x86_64-unknown-cygwin
12:51:42  checking host system type... x86_64-unknown-cygwin
12:51:58  error: module not found: jdk.hotspot.agent
12:52:08  checking whether the C compiler works... no
12:52:08  configure: error: in `/cygdrive/c/users/jenkins/workspace/build_jdknext_x86-64_windows_personal/build/windows-x86_64-server-release/vm/omr':
12:52:08  configure: error: C compiler cannot create executables
12:52:08  See `config.log' for more details
12:52:16  make[5]: *** [run_configure.mk:164: CONFIGURE_SENTINEL_FILE] Error 77
12:52:16  make[5]: Leaving directory '/cygdrive/c/users/jenkins/workspace/build_jdknext_x86-64_windows_personal/build/windows-x86_64-server-release/vm/omr'
12:52:16  make[4]: *** [/cygdrive/c/users/jenkins/workspace/build_jdknext_x86-64_windows_personal/openj9/runtime/buildtools.mk:180: configure] Error 2
```
It seems some VS compiler environment settings lost when invoking OMR configure.

**Comment by keithc-ca** [comment]
This should be solved by the combination of eclipse/openj9#11589, ibmruntimes/openj9-openjdk-jdk#263 and  ibmruntimes/openj9-openjdk-jdk16#3.

**Comment by pshipton** [comment]
The latest jdk16 and jdk17 content has been promoted now.
